lua/plenary/luascan.lua

```diff
@@ -0,0 +1,128 @@
+local m = {}
+
+local read_file = function(filepath)
+  local fd = vim.loop.fs_open(filepath, "r", 438)
+  if fd == nil then return '' end
+  local stat = assert(vim.loop.fs_fstat(fd))
+  if stat.type ~= 'file' then return '' end
+  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
+  assert(vim.loop.fs_close(fd))
+  return vim.split(data, '\n')
```

Comment:
For split by newline, it maybe faster to use this version. It is less fancy then vim.split `runtime/lua/shared.lua:63` but might slightly help with performance. I suggest hardcoding "\n" 

```lua
local split = function(s, sep)
  local done = false
  local acc = {}
  local index = 1
  if not sep then return {s} end
  while not done do
    local start, _end = string.find(s,sep,index)
    if type(start) == "nil" then 
      table.insert(acc, string.sub(s, index))
      done = true 
      else 
        table.insert(acc, string.sub(s, index, (start - 1)))
        index = _end + 1
    end
  end
  return acc
end

```

Reply:
Not for this but for grep. I was mentioning to tj that we kinda don't even have to split the files into lines. We could just write our own search string and keep track of how many `\n` we found. That would be the `lnum` and the chars since the last `\n` would be `col`.

But coming back to your example. It might be helpful to have a more performant split in plenary. If not for here then for `buffer_previewer`